optaplanner-core/src/main/java/org/optaplanner/core/api/score/stream/uni/UniConstraintStream.java

```diff
@@ -186,20 +186,109 @@
+     * {@link UniConstraintStream} which only has one item in it, the result of applying
+     * {@link UniConstraintCollector#finisher()}.
+     *
+     * @param collector never null, the collector to perform the grouping operation with
+     * @param <ResultContainer_> the mutable accumulation type (often hidden as an implementation detail)
+     * @param <Result_> the type of a match fact in the destination {@link UniConstraintStream}
+     * @return never null.
+     */
+    <ResultContainer_, Result_> UniConstraintStream<Result_> groupBy(
             UniConstraintCollector<A, ResultContainer_, Result_> collector);
 
-    <GroupKey_> UniConstraintStream<GroupKey_> groupBy(
-            Function<A, GroupKey_> groupKeyMapping);
-
+    /**
+     * Convert the {@link UniConstraintStream} to a different {@link UniConstraintStream}, consisting of the set of
+     * values resulting from applying the group key mapping function on all items in the original stream. None of the
+     * items in the new stream will {@link Objects#equals(Object, Object)} any other.
+     *
+     * @param groupKeyMapping never null, mapping function to convert each element in the stream to a different element.
+     * @param <GroupKey_> the type of a match fact in the destination {@link UniConstraintStream}.
+     * @return never null.
+     */
+    <GroupKey_> UniConstraintStream<GroupKey_> groupBy(Function<A, GroupKey_> groupKeyMapping);
+
+    /**
+     * Convert the {@link UniConstraintStream} to a {@link BiConstraintStream}, consisting of:
+     *
+     * <ul>
+     *     <li>on the left, the set of values resulting from applying the group key mapping function on all items in
+     *     the original stream. None of the items in the new stream will {@link Objects#equals(Object, Object)} any
+     *     other.</li>
+     *     <li>on the right, the value of applying a given {@link UniConstraintCollector} on all the items in the
+     *     original {@link UniConstraintStream} with a matching group key.</li>
+     * </ul>
+     *
+     * @param groupKeyMapping never null, mapping function to convert each element in the stream to a different element.
+     * @param <GroupKey_> the type of the left match fact in the destination {@link BiConstraintStream}.
+     * @param <ResultContainer_> the mutable accumulation type (often hidden as an implementation detail)
+     * @param <Result_> the type of the right match fact in the destination {@link BiConstraintStream}
+     * @return never null.
+     */
     <GroupKey_, ResultContainer_, Result_> BiConstraintStream<GroupKey_, Result_> groupBy(
             Function<A, GroupKey_> groupKeyMapping,
             UniConstraintCollector<A, ResultContainer_, Result_> collector);
 
+    /**
+     * Convert the {@link UniConstraintStream} to a {@link BiConstraintStream}, consisting of:
+     *
+     * <ul>
+     *     <li>on the left, the set of values resulting from applying the "A" group key mapping function on all items in
```

Comment:
We can't use "left" and "right" for BiStream here, because for TriStream due to a groupBy(getA, getB, getC), we can't use "left" and "right" either. (If the tristream is a join of A+B and C, then we can use "left" and "right").
We should use "A" or "first fact".